pwndbg/gdblib/config.py

```diff
@@ -0,0 +1,101 @@
+"""
+Dynamic configuration system for pwndbg, using GDB's built-in Parameter
+mechanism.
+
+To create a new pwndbg configuration point, create an instance of
+``pwndbg.lib.config.Parameter``.
+
+Parameters should be declared in the module in which they are primarily
+used, or in this module for general-purpose parameters.
+
+All pwndbg Parameter types are accessible via property access on this
+module, for example:
+
+    >>> pwndbg.gdblib.config.add_param('example-value', 7, 'an example')
+    >>> int(pwndbg.gdblib.config.example_value)
+    7
+"""
+import collections
+
+import gdb
+
+import pwndbg.decorators
+import pwndbg.lib.config
+
+config = pwndbg.lib.config.Config()
+
+# TODO: I don't think this needs to be an OrderedDict
+PARAM_CLASSES = collections.OrderedDict()
```

Comment:
Seems this can even be just:

```suggestion
PARAM_CLASSES = {
    bool: gdb.PARAM_BOOLEAN,
    int: gdb.PARAM_ZINTEGER,
    str: gdb.PARAM_STRING
}
```

Reply:
Yea, I just couldn't figure out why the original code used `OrderedDict`, and was worried about breaking something by changing it to a normal `dict`. But we can do it and see what happens.